numba/cuda/tests/cudadrv/test_cuda_array_slicing.py

```diff
@@ -308,6 +309,67 @@ def test_incompatible_shape(self):
                 "dimension 5",  # simulator
             ])
 
+    @skip_on_cudasim('cudasim does not use streams and operates synchronously')
+    def test_sync(self):
+        # There should be a synchronization when no stream is supplied
+        darr = cuda.to_device(np.arange(5))
+
+        with patch.object(cuda.cudadrv.driver.Stream, 'synchronize',
+                          return_value=None) as mock_sync:
+            darr[0] = 10
+
+        mock_sync.assert_called_once()
+
+    @skip_on_cudasim('cudasim does not use streams and operates synchronously')
+    def test_no_sync_default_stream(self):
+        # There should not be a synchronization when a the array has a default
+        # stream, whether it is the default stream, the legacy default stream,
```

Comment:
```suggestion
        # There should not be a synchronization when the array has a stream
        # provided, whether it is the default stream, the legacy default stream,
```

Reply:
I agree with the removal of "a" (that is an error), but saying "has a stream provided" seems less clear than "has a default stream", given that the default stream of a device array is a concept in Numba. However, this does highlight a deficiency in the docs for the stream constructors ([example](https://numba.readthedocs.io/en/stable/cuda-reference/memory.html#numba.cuda.device_array)), in that they don't explain that the stream parameter becomes the default stream for the array.

Reply:
(I'd like to follow this PR up today with another one that improves the docs for `device_array`, et al. so that they give a clear explanation of the "default stream".)

Reply:
Thanks for adding some more explanation, I agree that default stream needs to stay, but could do with some docs as noted.